src/lib/Microsoft.Health.Fhir.Ingest/Service/NormalizationStreamingEventConsumerService.cs

```diff
@@ -0,0 +1,142 @@
+using Microsoft.Health.Events.EventConsumers.Service;
+using Microsoft.Health.Events.Model;
+using Microsoft.Health.Fhir.Ingest.Data;
+using Microsoft.Health.Fhir.Ingest.Telemetry;
+using Microsoft.Health.Fhir.Ingest.Template;
+using Microsoft.Health.Logging.Telemetry;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Fhir.Ingest.Service
+{
+    public class NormalizationStreamingEventConsumerService : StreamingEventConsumerService
+    {
+        private readonly IContentTemplate _contentTemplate;
+        private readonly IConverter<IEventMessage, JObject> _converter;
+        private readonly IExceptionTelemetryProcessor _exceptionTelemetryProcessor;
+        private readonly IEnumerableAsyncCollector<IMeasurement> _collector;
+
+        public NormalizationStreamingEventConsumerService(
+            EventBatchingOptions options,
+            ICheckpointClient checkpointClient,
+            ITelemetryLogger log,
+            IEventProcessingMetricMeters eventProcessingMetricMeters,
+            IConverter<IEventMessage, JObject> converter,
+            IContentTemplate contentTemplate,
+            IExceptionTelemetryProcessor exceptionTelemetryProcessor,
+            IEnumerableAsyncCollector<IMeasurement> collector)
+            : base(options, checkpointClient, log, eventProcessingMetricMeters)
+        {
+            _contentTemplate = EnsureArg.IsNotNull(contentTemplate, nameof(contentTemplate));
+            _converter = EnsureArg.IsNotNull(converter, nameof(converter));
+            _exceptionTelemetryProcessor = EnsureArg.IsNotNull(exceptionTelemetryProcessor, nameof(exceptionTelemetryProcessor));
+            _collector = EnsureArg.IsNotNull(collector, nameof(collector));
+        }
+
+        /// <summary>
+        /// Reusable list to collect projected measurements
+        /// </summary>
+        private List<IMeasurement> NormalizationBatch { get; } = new List<IMeasurement>(50);
+
+        protected override async Task ConsumeEventImpl(IEventMessage eventArg)
+        {
+            try
+            {
+                RecordLatencyMetrics(eventArg);
+
+                var token = _converter.Convert(eventArg);
+
+                await NormalizeMessage(token, eventArg);
+            }
+            catch (Exception ex)
+            {
+                if (!_exceptionTelemetryProcessor.HandleException(ex, Logger))
+                {
+                    throw; // Immediately throw originalexception if it is not handled
+                }
+            }
+        }
+
+        private async Task NormalizeMessage(JObject token, IEventMessage eventArg)
+        {
+            Stopwatch sw = Stopwatch.StartNew();
+            foreach (var measurement in _contentTemplate.GetMeasurements(token))
+            {
+                try
+                {
+                    measurement.IngestionTimeUtc = eventArg.EnqueuedTime.UtcDateTime;
+                    NormalizationBatch.Add(measurement);

```

Comment:
Is this `NormalizationBatch` threadsafe? I ask because I think the previous `IEventConsumerService` implementations are singletons. So this would be used across partitions/threads.